graph/src/components/ethereum/network.rs

```diff
@@ -0,0 +1,273 @@
+use failure::{format_err, Error};
+use std::cmp::Ord;
+use std::collections::{BTreeMap, HashMap};
+use std::fmt;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use crate::components::ethereum::EthereumAdapter;
+pub use crate::impl_slog_value;
+
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub enum NetworkCapability {
```

Comment:
I'm strongly tempted to suggest that `(Ethereum)NodeCapability` is the better name. Everywhere this shows up it feels weird to me to talk about the network, when there may be other adapters for the same network that _have_ the right capability. So it's per node/provider, not per network.

Reply:
I like that idea, I have been starting to feel that NetworkCapability is confusing as well.  
I find that when I'm writing about these components I'm unsure whether to use "network", "endpoint", or "node". 
I'll update it to be `NodeCapability`; since it's in the `ethereum` component I don't think we need to include that in the name.